### ISSUE TITLE
Migrating to the new API

### ISSUE THREAD

**Issue by LorenzoBettini** [hcoles/pitest#1283]
Hi

In Pitclipse, https://github.com/pitest/pitclipse, we need to update to a more recent version of pit to support Java 21 (https://github.com/pitest/pitclipse/issues/221). We were using 1.6.8 and moving to 1.15.3 leads to compilation errors. I seem to understand this is due to https://github.com/hcoles/pitest/commit/95e8102725b59780c07e89d4fca9a7563f12c976 which changes the API of `CoverageDatabase` (we were using its methods `getNumberOfCoveredLines` and `getClassInfo`, which now are gone), which we used to retrieve from `ListenerArguments.getCoverage`, which now returns a `ReportCoverage`.

Is there any suggested way to migrate to the new API?

thanks in advance

**Comment by hcoles** [comment]
Yes, the interface as changed a lot and the ClassInfo class has been largely deprecated.

You should have all the data you need to determine the number of covered lines for the

```java
Set<ClassLine> getCoveredLines(ClassName clazz);
```

Method.

How were you using ClassInfo?

**Comment by LorenzoBettini** [comment]
@hcoles thanks for the answer.

That's the main part https://github.com/pitest/pitclipse/blob/0b1f2926d97fb0b92f3c79dd31f4667823d287c5/bundles/org.pitest.pitclipse.runner/src/org/pitest/pitclipse/runner/results/summary/SummaryResultListener.java#L63 where we are using `CoverageDatabase`.

Concerning `ClassInfo` then we use it mainly here: https://github.com/pitest/pitclipse/blob/0b1f2926d97fb0b92f3c79dd31f4667823d287c5/bundles/org.pitest.pitclipse.runner/src/org/pitest/pitclipse/runner/results/summary/ClassSummary.java#L46

We are also using `MethodName`, which I seem to understand it's now removed; but IIRC we use that in tests, so that's less crucial.
